modules/kubernetes/ingress-nginx/main.tf

```diff
@@ -29,6 +29,133 @@ resource "kubernetes_namespace" "this" {
   }
 }
 
+resource "kubernetes_network_policy" "deny_default" {
+  metadata {
+    name      = "deny-default"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {}
+    }
+
+    policy_types = ["Ingress", "Egress"]
+
+    ingress {
+      from {
+        pod_selector {}
+      }
+    }
+
+    egress {
+      to {
+        namespace_selector {}
+        pod_selector {
+          match_labels = {
+            k8s-app = "kube-dns"
+          }
+        }
+      }
+
+      ports {
+        port     = 53
+        protocol = "UDP"
+      }
+    }
+
+    egress {
+      to {
+        pod_selector {}
+      }
+    }
+  }
+}
+
+resource "kubernetes_network_policy" "allow_scraping" {
+  metadata {
+    name      = "allow-scraping"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {
+        "app.kubernetes.io/component" = "controller"
+        "app.kubernetes.io/instance" = "ingress-nginx"
+        "app.kubernetes.io/name" = "ingress-nginx"
+      }
+    }
+
+    policy_types = ["Ingress"]
+
+    ingress {
+      from {
+        namespace_selector {
+          match_labels = {
+            name = "prometheus"
+          }
+        }
+        pod_selector {
+          match_labels = {
+            app = "prometheus"
+            "app.kubernetes.io/name" = "prometheus"
+          }
+        }
+      }
+
+      ports {
+        port = "metrics"
+        protocol = "TCP"
+      }
+    }
+  }
+}
+
+resource "kubernetes_network_policy" "allow_ingress_traffic" {
+  metadata {
+    name      = "allow-ingress-traffic"
+    namespace = kubernetes_namespace.this.metadata[0].name
+  }
+
+  spec {
+    pod_selector {
+      match_labels = {
+        "app.kubernetes.io/component" = "controller"
+        "app.kubernetes.io/instance" = "ingress-nginx"
+        "app.kubernetes.io/name" = "ingress-nginx"
+      }
+    }
+
+    policy_types = ["Ingress", "Egress"]
+
+    ingress {
+      from {
+        ip_block {
+          cidr = "0.0.0.0/0"
+          except = [
+            "10.0.0.0/8",
+            "172.16.0.0/12",
+            "192.168.0.0/16",
+          ]
```

Comment:
Could this cause issues for internal ingress?

Reply:
Yes we should add some flag to flip the two CIDRs with a flag if it is a private ingress controller.

Reply:
We had to remove this as internal traffic hitting the ingress controller shows up as coming from an internal IP.